api/opds.py

```diff
@@ -781,13 +781,29 @@ def drm_device_registration_feed_tags(self, patron):
             tag.attrib[attr] = "http://librarysimplified.org/terms/drm/scheme/ACS"
         return tags
 
+    @property
+    def user_profile_management_protocol_link(self):
+        """Create a <link> tag that points to the circulation
+        manager's User Profile Management Protocol endpoint
+        for the current patron.
+        """
+        link = OPDSFeed.makeelement("link")
+        link.attrib['rel'] = 'http://librarysimplified.org/terms/rel/user-profile'
+        link.attrib['href'] = self.url_for(
+            'patron_profile', _external=True
+        )
+        return link
+        
     def annotate_feed(self, feed, lane):
-        """Add feed-level DRM device registration tags to the feed."""
+        """Annotate the feed with top-level DRM device registration tags
+        and a link to the User Profile Management Protocol endpoint.
+        """
         super(CirculationManagerLoanAndHoldAnnotator, self).annotate_feed(
             feed, lane
```

Comment:
Lane is always going to be None in this situation because this is the patron's loans feed. But it's part of the method signature because annotate_feed() is part of the general OPDS feed interface, and most feeds are in the context of a specific lane.